File: C++Projects/POO_3964_Clase03_HerenciaMultipleYPackage/ParentClasses/Mamifero.cpp
//
// Created by Diego Cruz on 19/03/2024.
//

#include "Mamifero.h"

// Constructor
Mamifero::Mamifero(string nombre) {
    this->nombre = nombre;
}

// Getter y Setter
string Mamifero::getNombre() {
    return nombre;
}
void Mamifero::setNombre(string nombre) {
    Mamifero::nombre = nombre;
}

File: JavaProjects/POO_3964_Clase03/src/Mamifero.java
public class Mamifero {
    // Propiedad dada por encapsulamiento
    private String nombre;
    // Constructor
    public Mamifero(String nombre){
        this.nombre = nombre;
    }
    public Mamifero(){
        this.nombre = "Por definir";
    }

    // Método para todos los mamíferos
    public void Expresarse(){
        System.out.println("Por definir");
    }

    // Setters y Getters
    public String getNombre() {
        return nombre;
    }
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
}
